### PR TITLE
Remove DevicePermissionDescriptor deviceId.

### PR DESCRIPTION
**Pull request by jan-ivar**
Fixes https://github.com/w3c/mediacapture-main/issues/965.

***
<a href="https://pr-preview.s3.amazonaws.com/jan-ivar/mediacapture-main/pull/968.html" title="Last updated on Aug 17, 2023, 4:55 PM UTC (c7529b7)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/mediacapture-main/968/76a2643...jan-ivar:c7529b7.html" title="Last updated on Aug 17, 2023, 4:55 PM UTC (c7529b7)">Diff</a>

### CODE REVIEW

File: getusermedia.html [hunk 1]
```diff
@@ -5427,34 +5425,29 @@ <h1 id="permissions-integration">Permissions Integration</h1>
...
+          Thus, if a query
+          for the <a>"camera"</a> permission returns {{PermissionState/"granted"}}, the
           client knows that it will get access to one camera without a permission prompt, and if
           {{PermissionState/"denied"}} is returned, it knows that no getUserMedia request for a
           camera will succeed.
         </p>
         <p>
-          If a {{PermissionState/"granted"}} permission is present on some, but not all, devices of a kind, a query
-          without the {{DevicePermissionDescriptor/deviceId}} will return
+          If the User agent considers permission given to some, but not all, devices of a kind, a query
```
Comment (jan-ivar) [inline getusermedia.html:5444]:
```suggestion
          If the User Agent considers permission given to some, but not all, devices of a kind, a query
```

File: getusermedia.html [hunk 2]
```diff
@@ -5463,44 +5456,17 @@ <h1 id="permissions-integration">Permissions Integration</h1>
...
-      </dd>
       <dt>
         [=powerful feature/permission revocation algorithm=]
       </dt>
       <dd>
         This is the result of calling the [=device permission revocation algorithm=] passing
-        {{PermissionDescriptor/name}} and {{DevicePermissionDescriptor/deviceId}} as arguments.
-        If the descriptor does not have a {{DevicePermissionDescriptor/deviceId}}, then
-        <code>undefined</code> is passed in place of {{DevicePermissionDescriptor/deviceId}}.
+        {{PermissionDescriptor/name}} and <code>undefined</code> as arguments.
```
Comment (jan-ivar) [inline getusermedia.html:5469]:
```suggestion
        {{PermissionDescriptor/name}} as argument.
```

**Review by youennf — APPROVED** [review]

**Comment by youennf** [comment]
Waiting for @alvestrand input before merging.

File: getusermedia.html [hunk 3]
```diff
@@ -882,7 +882,7 @@ <h4>Life-cycle</h4>
         <p>If the end of the track was reached due to a user request, the event
         source for this event is the user interaction event source.</p>
         <p>To invoke the <dfn class="abstract-op">device permission revocation algorithm</dfn> with <var>permissionName</var> and
```
Comment (jan-ivar) [inline getusermedia.html:884]:
```suggestion
        <p>To invoke the <dfn class="abstract-op">device permission revocation algorithm</dfn> with <var>permissionName</var>,
```

File: getusermedia.html [hunk 4]
```diff
@@ -882,7 +882,7 @@ <h4>Life-cycle</h4>
...
-        {{DevicePermissionDescriptor/deviceId}} <var>deviceId</var>, run the
+        {{MediaDeviceInfo/deviceId}} <var>deviceId</var>, run the
```
Comment (jan-ivar) [inline getusermedia.html:885]:
```suggestion
        run the
```

Comment (alvestrand) [reply getusermedia.html]: Need to take out the language at line 894 too.

**Pull request merged**